.github/workflows/bench.yml

```diff
@@ -0,0 +1,86 @@
+name: BENCHMARK
+
+# Controls when the action will run.
+on:
+  pull_request:
+    branches: testnet
+
+jobs:
+  hardware-benchmarks:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env
+        run: |
+          echo "HOME=/home/actions" >> $GITHUB_ENV
+          pallet=$(grep "impl pallet_" runtime/src/lib.rs | grep "Config for Runtime" | sed 's/impl pallet_//' | cut -f1 -d: | sort)
+          for i in $pallet; do
+            mkdir -p pallets/$i/src
+          done
```

Comment:
Here you are creating directories for all available pallets automatically but down below each pallet is configured individually and you can also move the mkdir command there as well. This section is not very useful as it is.